paddle/gserver/gradientmachines/RecurrentGradientMachine.cpp

```diff
@@ -1399,8 +1399,8 @@ void RecurrentGradientMachine::createDataOutlinkCopySizeInfo(
              getBeamSize() > 1 ? finalPaths_.size() : finalPaths_[0].size());
     int* starts = inputSeqStartPos->getMutableData(false);
     int seqId = 0;
-    for (int i = 0; i < finalPaths_.size(); ++i) {
-      for (int j = 0; j < finalPaths_[i].size(); ++j) {
+    for (size_t i = 0; i < finalPaths_.size(); ++i) {
+      for (size_t j = 0; j < finalPaths_[i].size(); ++j) {
```

Comment:
@Canpio I think we should not obey google c++ style for this situation.

Because all STL containers are rewritten in google. They make containers' size_type as `int`. We are using `std`'s containers like `unordered_map`, `vector`, which size_type is `size_t`.

So maybe just use size_t is fine. Using `int` only has an advantage when reversely access a container,i.e., when uses `--` operator.